File: Java/src/test/java/com/gildedrose/TexttestFixture.java
package com.gildedrose;

import org.approvaltests.Approvals;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;

public class TexttestFixture {
    public static void main(String[] args) {
        System.out.println("OMGHAI!");

        Item[] items = new Item[] {
                new Item("+5 Dexterity Vest", 10, 20), //
                new Item("Aged Brie", 2, 0), //
                new Item("Elixir of the Mongoose", 5, 7), //
                new Item("Sulfuras, Hand of Ragnaros", 0, 80), //
                new Item("Sulfuras, Hand of Ragnaros", -1, 80),
                new Item("Backstage passes to a TAFKAL80ETC concert", 15, 20),
                new Item("Backstage passes to a TAFKAL80ETC concert", 10, 49),
                new Item("Backstage passes to a TAFKAL80ETC concert", 5, 49),
                // this conjured item does not work properly yet
                new Item("Conjured Mana Cake", 3, 6) };

        GildedRose app = new GildedRose(items);

        int days = 2;
        if (args.length > 0) {
            days = Integer.parseInt(args[0]) + 1;
        }

        for (int i = 0; i < days; i++) {
            System.out.println("-------- day " + i + " --------");
            System.out.println("name, sellIn, quality");
            for (Item item : items) {
                System.out.println(item);
            }
            System.out.println();
            app.updateQuality();
        }
    }

    private java.io.PrintStream originalSysOut;
    private java.io.ByteArrayOutputStream capturedStream;

    @Before
    public void setUp() throws Exception {

        this.originalSysOut = System.out;
        this.capturedStream = new java.io.ByteArrayOutputStream();
        java.io.PrintStream printStream = new java.io.PrintStream(this.capturedStream);
        System.setOut(printStream);
    }

    @Test
    public void goldenMasterTest() {
        TexttestFixture.main(new String[0]);
        Approvals.verify(capturedStream);
    }

    @After
    public void tearDown() throws Exception {
        System.setOut(this.originalSysOut);
    }
}

File: Kotlin/src/test/kotlin/com/gildedrose/TexttestFixture.kt
package com.gildedrose

import org.approvaltests.Approvals
import org.junit.After
import org.junit.Before
import org.junit.Test

fun main(args: Array<String>) {

    println("OMGHAI!")

    val items = arrayOf(createItem("+5 Dexterity Vest", 10, 20), //
            createItem("Aged Brie", 2, 0), //
            createItem("Elixir of the Mongoose", 5, 7), //
            createItem("Sulfuras, Hand of Ragnaros", 0, 80), //
            createItem("Sulfuras, Hand of Ragnaros", -1, 80),
            createItem("Backstage passes to a TAFKAL80ETC concert", 15, 20),
            createItem("Backstage passes to a TAFKAL80ETC concert", 10, 49),
            createItem("Backstage passes to a TAFKAL80ETC concert", 5, 49),
            // this conjured item does not work properly yet
            createItem("Conjured Mana Cake", 3, 6))

    val app = GildedRose(items)

    var days = 2
    if (args.size > 0) {
        days = Integer.parseInt(args[0]) + 1
    }

    for (i in 0..days - 1) {
        println("-------- day $i --------")
        println("name, sellIn, quality")
        for (item in items) {
            println(item)
        }
        println()
        app.updateQuality()
    }
}



class TestGoldenMaster {

    private val originalSysOut: java.io.PrintStream = System.out
    private var capturedStream: java.io.ByteArrayOutputStream = java.io.ByteArrayOutputStream()

    @Before
    @Throws(Exception::class)
    fun setUp() {
        this.capturedStream = java.io.ByteArrayOutputStream()
        val printStream = java.io.PrintStream(this.capturedStream)
        System.setOut(printStream)
    }

    @Test
    fun itShouldGoldenMaster() {
        main(emptyArray())
        Approvals.verify(capturedStream);
    }

    @After
    @Throws(Exception::class)
    fun tearDown() {
        System.setOut(this.originalSysOut)
    }
}